mono/utils/mono-context.h

```diff
@@ -160,7 +209,8 @@ typedef struct {
 
 typedef struct {
 	mgreg_t gregs [AMD64_NREG];
-	double fregs [AMD64_XMM_NREG];
+    /* See note: [MonoContextSimdReg Alignment] */
+	MonoContextSimdReg fregs [AMD64_XMM_NREG];
```

Comment:
Did you consider adding a separate array to hold the XMM regs?
That would mean overlap of values, but it would avoid changing a lot of low level code in the JIT.